modules/nf-core/toulligqc/main.nf

```diff
@@ -25,15 +26,13 @@ process TOULLIGQC {
     def args = task.ext.args ?: ''
     def prefix = task.ext.prefix ?: "${meta.id}"
 
-    def seq_summary_input = seq_summary ? "--sequencing-summary-source ${seq_summary}" : ""
-    def fastq_input = fastq ? "--fastq ${fastq}" : ""
-    def bam_input = bam ? "--bam ${bam}" : ""
+    def input_file = ("$ontfile".endsWith(".fastq") || "$ontfile".endsWith(".fastq.gz") || "$ontfile".endsWith(".fq") || "$ontfile".endsWith(".fq.gz")) ? "--fastq ${ontfile}" :
+        ("$ontfile".endsWith(".txt") || "$ontfile".endsWith(".txt.gz")) ? "--sequencing-summary-source ${ontfile}" :
+        ("$ontfile".endsWith(".bam")) ? "--bam ${ontfile}" : ''
```

Comment:
Would this work with cram input?
Should this fail at the end?

Reply:
No, ToulligQC doesn't deal with CRAM files.
If no files/ incorrect files are given, the module will fail yes!